build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/order/ByFullName.java

```diff
@@ -3,11 +3,62 @@
 import hudson.model.Job;
 
 import java.io.Serializable;
+import java.util.Arrays;
 import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class ByFullName implements Comparator<Job<?, ?>>, Serializable {
+
+    private OrdinalSet ordinalSet;
+
+    public OrdinalSet getOrdinalSet() {
+        return ordinalSet;
+    }
+
+    public void setOrdinalSet(OrdinalSet ordinalSet) {
+        this.ordinalSet = ordinalSet;
+    }
+
     @Override
     public int compare(Job<?, ?> a, Job<?, ?> b) {
+        if (ordinalSet != null) {
+            return ordinalSet.expandWithOrdinalNumber(a.getFullName()).compareToIgnoreCase(
+                    ordinalSet.expandWithOrdinalNumber(b.getFullName())
+            );
+        }
         return a.getFullName().compareToIgnoreCase(b.getFullName());
     }
+
+    public static class OrdinalSet {
+        private final List<String> entries;
+
+        public OrdinalSet(List<String> entries) {
+            this.entries = entries;
+        }
+
+        public List<String> getEntries() {
+            return entries;
+        }
+
+        String expandWithOrdinalNumber(String name) {
+            if (entries != null) {
+                for(int i = 0; i< entries.size(); i++) {
+                    name = name.replaceAll(entries.get(i), "#" + i + "_" + entries.get(i));
```

Comment:
the concept of having a string for the comparator is not very robust.
we will use jobs with different folders, so we definitely need to include foldernames in the ordinalset in order for this to work.

shouldn't we rather use `.contains` and return the index number, when there is a match?

Reply:
I did not understand your proposal. Please be more precise, how you would change the implementation..

I thought about just looping the ordinal sets until the first match, and do the replacement just there, without further looping... so this would have been done with ".contains" and simply do one replacement. Don't know if this was in your mind.

Reply:
```java
for (int i = 0; i < entries.size(); i++) {
    if (entries.get(i).contains(name)) {
        return i;
    }
}
```

Reply:
Having a look at the example mentioned in the help-ordinalSet.html file:
https://github.com/cgiorgetta/jenkins-build-monitor-plugin/blob/feature/advanced_view_control/build-monitor-plugin/src/main/resources/com/smartcodeltd/jenkinsci/plugins/buildmonitor/BuildMonitorView/help-ordinalSet.html

... if you would just return the ordinalSet position, then this would result in having first all "CI_Pipeline" jobs, then all "Deploy_to__Test" jobs, and finally the "Deploy_to_Prod" job. I.e. the jobs would not be clustered by applications, but by job type, which is not what we want.